include/misc/dlist.h

```diff
@@ -552,6 +552,68 @@ static inline sys_dnode_t *sys_dlist_get(sys_dlist_t *list)
 	return node;
 }
 
+/**
+ * @brief place the contents of one list at the end of another list.
+ *
+ * The @p to and @p from lists must be distinct.  On completion @p from
+ * will be empty, all of its elements having been appended in original
+ * order to @p to.
+ *
+ * @param to a list, possibly non-empty, to which from will be appended
+ * @param from the list providing the elements to append
+ *
+ * @return N/A
+ */
+static inline void sys_dlist_catenate(sys_dlist_t *to,
```

Comment:
Strictly speaking the word is `catenate`: to join.  `concatenate` is to join to self.  Could do `sys_dlist_join`.  Least appealing is `concat` which would be like saying `sys_dlist_app` instead of `sys_dlist_append`.

I like the bike shed as it is, but further opinions are welcome.

Reply:
Join is ok for me. You are right that this is kind of a bike sched, but I had to google the "catenate" world and looked for something which would not require such action.